aws-blog-event-driven-batch-analytics/src/main/java/com/amazonaws/bigdatablog/edba/LambdaContainer.java

```diff
@@ -0,0 +1,249 @@
+				String jobId = fireEMRJob(rs.getString("job_params"),clusterId);
+				updateJobConfigPS.setString(1,clusterId+":"+jobId);
+				updateJobConfigPS.setString(2, rs.getString("job_config_id"));
+				updateJobConfigPS.addBatch();
+				updateSubmittedJobsStmt.addBatch(updateSubmittedJobsJSON.replaceAll("\\?", rs.getString("job_config_id"))+rs.getString("job_input_pattern"));
+
+			}
+		}
+		updateJobConfigPS.executeBatch();
+		updateSubmittedJobsStmt.executeBatch();
+		updateSubmittedJobsStmt.close();
+		updateJobConfigPS.close();
+		conditionFetchStmt.close();
+		conn.close();
+	}
+	
+	// EMR Job Monitor lambda function
+	
+	public void monitorEMRStep() throws Exception {
+//		Class.forName("com.mysql.jdbc.Driver").newInstance();
+		List<String> stepIds = new ArrayList<String>();
+		Connection conn = DriverManager.getConnection(url, user, password);
+		ResultSet openStepsRS = conn.createStatement().executeQuery(retrieveOpenSteps);
+		AmazonElasticMapReduceClient emr = new AmazonElasticMapReduceClient();
+		DescribeStepRequest stepReq=new  DescribeStepRequest();
+		PreparedStatement ps = conn.prepareStatement(updateStepStatus);
+		while(openStepsRS.next()){
+			
+			stepReq.setClusterId(openStepsRS.getString("cluster_id"));
+			stepReq.setStepId(openStepsRS.getString("step_id"));
+			String stepState = emr.describeStep(stepReq).getStep().getStatus().getState();
+			
+				if(stepState.equals(StepState.COMPLETED.toString())){
+					ps.setString(1,StepState.COMPLETED.toString());
+				}else if (stepState.equals(StepState.FAILED.toString())){
+					ps.setString(1,StepState.FAILED.toString());					
+				}
+				ps.setString(2,openStepsRS.getString("job_config_id"));
+				ps.addBatch();
+		}
+		
+		ps.executeBatch();
+		ps.close();
+		conn.close();
+	}
+	
+	// adds EMR step
+	protected String fireEMRJob(String paramsStr,String clusterId){
+		StepFactory stepFactory = new StepFactory();
+		AmazonElasticMapReduceClient emr = new AmazonElasticMapReduceClient();
+		emr.setRegion(Region.getRegion(Regions.US_EAST_1));
+		Application sparkConfig = new Application()
+				.withName("Spark");
+		
+		String[] params = paramsStr.split(",");
+		StepConfig enabledebugging = new StepConfig()
+				.withName("Enable debugging")
+				.withActionOnFailure("TERMINATE_JOB_FLOW")
+				.withHadoopJarStep(stepFactory.newEnableDebuggingStep());
+		
+		HadoopJarStepConfig sparkStepConf = new HadoopJarStepConfig()
+				.withJar("command-runner.jar")
+				.withArgs(params);	
+		
+		final StepConfig sparkStep = new StepConfig()
+				.withName("Spark Step")
+				.withActionOnFailure("CONTINUE")
+				.withHadoopJarStep(sparkStepConf);
+
+		
+		AddJobFlowStepsRequest request = new AddJobFlowStepsRequest(clusterId)
+				.withSteps(new ArrayList<StepConfig>(){{add(sparkStep);}});
+				
+
+		AddJobFlowStepsResult result = emr.addJobFlowSteps(request);
+		return result.getStepIds().get(0);
+	}
+	
+	protected List<String> getActiveClusterId() throws Exception{
```

Comment:
I think this method should be scoped to clusters that are tagged in a certain way, by this application. Otherwise you wait on **all** clusters in the customer's account that are visible to the Role.